src/App.tsx

```diff
@@ -124,12 +129,26 @@ function App() {
     }
   };
 
+  const setExtensionIdToLocalStorage = useCallback(
+    (_extensionId: string): void => {
+      const walletList = Object.values(walletMapEntries);
+      const detectedExtensionIds = walletList.map(
+        (entry) => entry.params.extensionId,
+      );
+      if (detectedExtensionIds.includes(_extensionId)) {
+        return;
+      }
+      localStorage.setItem('extensionId', _extensionId);
+    },
+    [walletMapEntries],
+  );
+
   const handleConnect = () => {
-    if (extensionId && provider) {
+    if (provider && extensionId) {
```

Comment:
does this reorder matter?

Reply:
LOL! this was me removing `extensionId` from that verification, and then slapping it back in 😅 one does not notice because I rebase all my commits after the work for clean up, but that's the reasoning for that apparent re-order.